compiler/data/kphp-json-tags.cpp

```diff
@@ -0,0 +1,229 @@
+    {"raw_string",        [](auto &this_, vk::string_view  ) { this_.raw_string_ = true; }, false}
+  };
+  bool match = run_parse_tag(known_tags, tag, *this);
+  kphp_error(match, fmt_format("unknown @kphp-json tag '{}' over class field", tag));
+}
+
+ClassTags::ClassTags(const PhpDocComment &phpdoc) noexcept {
+  for (const PhpDocTag &tag : phpdoc.tags) {
+    parse_tag(tag.value);
+  }
+}
+
+void ClassTags::parse_tag(vk::string_view tag) noexcept {
+  const static std::initializer_list<KnownTag_<ClassTags>> known_tags{
+    {"skip_if_default",   [](auto &this_, vk::string_view v) { this_.has_skip_if_default_ = parse_skip_if_default(v); }},
+    {"fields_rename",     [](auto &this_, vk::string_view v) { this_.policy_ = parse_fields_rename_policy(v); }},
+    {"fields_visibility", [](auto &this_, vk::string_view v) { this_.fields_visibility_ = parse_fields_visibility(v); }},
+    {"float_precision",   [](auto &this_, vk::string_view v) { this_.float_precision_ = parse_float_precision(v); }},
+    {"flatten",           [](auto &this_, vk::string_view  ) { this_.has_flatten_ = true; }, false}
+  };
+  bool match = run_parse_tag(known_tags, tag, *this);
+  kphp_error(match, fmt_format("unknown @kphp-json tag '{}' above class", tag));
+}
+
+EncoderTags::EncoderTags(const ClassData &json_encoder) noexcept {
+  set_skip_if_default(json_encoder);
+  set_fields_rename(json_encoder);
+  set_fields_visibility(json_encoder);
+  set_float_precision(json_encoder);
+}
+
+void EncoderTags::set_skip_if_default(const ClassData &json_encoder) noexcept {
+  const auto *skip_if_default = json_encoder.get_constant("skip_if_default");
+  if (!skip_if_default || skip_if_default->value.try_as<op_false>()) {
+    return;
+  }
+  auto value = skip_if_default->value.try_as<op_true>();
+  kphp_error(value, "'skip_if_default' constant must be a bolean");
+  has_skip_if_default_ = true;
+}
+
+void EncoderTags::set_fields_rename(const ClassData &json_encoder) noexcept {
+  if (const auto *fields_rename = json_encoder.get_constant("fields_rename")) {
+    auto value = fields_rename->value.try_as<op_string>();
+    kphp_error(value, "'fields_rename' constant must be a string");
+    policy_ = parse_fields_rename_policy(value->get_string());
+  }
+}
+
+void EncoderTags::set_fields_visibility(const ClassData &json_encoder) noexcept {
+  if (const auto *fields_visibility = json_encoder.get_constant("fields_visibility")) {
+    auto value = fields_visibility->value.try_as<op_string>();
+    kphp_error(value, "'fields_visibility' constant must be a string");
+    fields_visibility_ = parse_fields_visibility(value->get_string());
+  }
+}
+
+void EncoderTags::set_float_precision(const ClassData &json_encoder) noexcept {
+  if (const auto *float_precision = json_encoder.get_constant("float_precision")) {
+    auto value = float_precision->value.try_as<op_int_const>();
+    kphp_error(value, "'float_precision' constant must be integer");
```

Comment:
`const float_precision` should be an integer

Reply:
same as above